test/cppsuite/test_harness/workload/database_operation.cpp

```diff
@@ -380,9 +380,14 @@ database_operation::update_operation(thread_context *tc)
 }
 
 void
-database_operation::validate(const std::string &operation_table_name,
+database_operation::validate(bool custom_tracking_table, const std::string &operation_table_name,
```

Comment:
This is the method the user will override, correct? It is strange to have an input parameter to indicate if the method is overridden or not.

Reply:
This is used here because we have two different things we can customise: The tracking table and the validation logic. Custom validation is implemented by overriding this function, but the custom table schema is defined separately in the config.

The flag is only needed in the base function as it allows us to detect a user who is trying to use the default validation, but has changed the tracking table schema which will likely cause issues.

That said, I'll play around with a few ideas and see if we can make this cleaner.

Reply:
I am thinking of the following @ajmorton:

- Keep the current `validate` function exclusive to the base class.
- Create a new function with no implementation but `virtual`.

When we want to perform validation, we check for the config item and we call one or the other.

Reply:
If we no longer need `is_custom`, we can keep that function and remove the first arg.

Reply:
`is_custom` has been removed and replaced with the checks I described above in 325247ff2f0d2cdd6175e87c536d66b735e51806

Reply:
Changed in 325247ff2f0d2cdd6175e87c536d66b735e51806 as per [this comment](https://github.com/wiredtiger/wiredtiger/pull/7872#discussion_r863267805)